### PR TITLE
Implement create group account process

### LINKED ISSUE #34: Implement MsgCreateGroupAccountStd
* MsgCreateGroupAccountStd (Critical Path)
As an admin of a group I want to create an account address for a group with a decision policy so that the group can manage tokens.
I want to reuse an existing group as electorate for different accounts.

*Q: Can only the admin of a group create an account or is it open to everybody?*

needed-by #31

### PR DESCRIPTION
**Pull request by alpe**
Resolves #34

Implements process with the assumptions:

* Only the admin of a group can create an account
* Weave Conditions to generate an account address

Merge after #51

### CODE REVIEW

File: incubator/group/integration_test.go [hunk 1]
```diff
@@ -97,9 +96,10 @@ func TestCreateGroupScenario(t *testing.T) {
 	var seq uint64
 	for msg, spec := range specs {
 		t.Run(msg, func(t *testing.T) {
-			msgs := []sdk.Msg{spec.src}
-			privs, accNums, seqs := []crypto.PrivKey{myKey}, myAccount.GetAccountNumber(), myAccount.GetSequence()
-			tx := types.NewTestTx(ctx, msgs, privs, []uint64{accNums}, []uint64{seqs}, fee)
+			accSeq, err := app.AccountKeeper.GetSequence(ctx, myAddr)
+			require.NoError(t, err)
+			privs, accNums := []crypto.PrivKey{myKey}, myAccount.GetAccountNumber()
```
Comment (anilcse) [inline incubator/group/integration_test.go]: Can you move this outside of the loop.

File: incubator/group/keeper.go [hunk 2]
```diff
@@ -257,12 +257,17 @@ func (k Keeper) setParams(ctx sdk.Context, params Params) {
 func (k Keeper) CreateGroupAccount(ctx sdk.Context, admin sdk.AccAddress, groupID GroupID, policy ThresholdDecisionPolicy, comment string) (sdk.AccAddress, error) {
 	maxCommentSize := k.MaxCommentSize(ctx)
 	if len(comment) > maxCommentSize {
```
Comment (anilcse) [inline incubator/group/keeper.go:259]: Can we keep it as part of `ValidateBasic()` ?

File: incubator/group/integration_test.go [hunk 3]
```diff
@@ -112,7 +112,108 @@ func TestCreateGroupScenario(t *testing.T) {
...
+		},
+	}
+
+	var seq uint64
+	for msg, spec := range specs {
+		t.Run(msg, func(t *testing.T) {
+			msgs := []sdk.Msg{spec.src}
+			accSeq, err := app.AccountKeeper.GetSequence(ctx, myAddr)
+			require.NoError(t, err)
+			privs, accNums := []crypto.PrivKey{myKey}, myAccount.GetAccountNumber()
```
Comment (anilcse) [inline incubator/group/integration_test.go]: Same as above

**Review by anilcse — COMMENTED** [review]
lgtm
Just a couple of comments, please fix them.

File: incubator/group/keeper.go (hunk 2 above)
Comment (anilcse) [reply incubator/group/keeper.go:259]: Nevermind, it requires paramSpace access. Fits here.

File: incubator/group/integration_test.go [hunk 4]
```diff
@@ -112,7 +112,108 @@ func TestCreateGroupScenario(t *testing.T) {
...
+	_, _, otherAddr := types.KeyTestPubAddr()
+
+	balances := sdk.NewCoins(sdk.NewInt64Coin("atom", 10000))
+	require.NoError(t, app.BankKeeper.SetBalances(ctx, myAddr, balances))
+
+	myGroupID, err := app.GroupKeeper.CreateGroup(ctx, myAddr, nil, "integration test")
+	require.NoError(t, err)
+
+	fee := types.NewTestStdFee()
+	specs := map[string]struct {
```
Comment (anilcse) [inline incubator/group/integration_test.go:132]: Can you add one (or more) test (s) to cover custom `Threshold` and `Timeout`s

Comment (alpe) [reply incubator/group/integration_test.go:132]: They are covered in unit test: https://github.com/regen-network/cosmos-modules/pull/52/files#diff-7b4842d86e976eaae9d5f8ac29447b16R192

Integration tests setup a full stack and are quite slow compared to the unit tests. That's why I try to keep them to the bare minimum.

**Review by anilcse — APPROVED** [review]
utACK :tada:

**Pull request merged**